### PR TITLE
Bump Antrea to Dragonfish.1 (1.2.3)

### PR DESCRIPTION
**Pull request by knabben**
### What this PR does / why we need it
Bumping Antrea Dragonfish.1

### Which issue(s) this PR fixes

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/1027

### Release note

```release-note
Antrea 1.2.3 support for ClusterResourceSet
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by knabben** [comment]
//usebom:sandbox/v1.5.0-zshippable/tkg-compatibility

**Comment by knabben** [comment]
//usebom:

**Comment by knabben** [comment]
//usebom

**Comment by vuil** [comment]
@knabben usebom directive is meant to instruct the CI to build and test using an alternative BOM.
An example is :
https://github.com/vmware-tanzu/tanzu-framework/pull/501#issuecomment-904196739

**Review by edwardbadboy — COMMENTED** [review]
Maybe a dumb question: In which file of this patch I could confirm that it's using the build https://buildweb.eng.vmware.com/ob/18841399/ ?

**Comment by knabben** [comment]
> Maybe a dumb question: In which file of this patch I could confirm that it's using the build https://buildweb.eng.vmware.com/ob/18841399/ ?

This is using the upstream configuration and the reason it is different from the build. I'm changing it accordingly.

File: pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml
Comment (knabben) [reply pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml:221]: thanks for catching this one. updating both places

File: pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml
Comment (knabben) [reply pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml:34]: Changing them on kapp-config, can you confirm these are the expected ones?
```
4179-  name: v1alpha1.stats.antrea.io
4194-  name: v1alpha1.stats.antrea.tanzu.vmware.com
4209-  name: v1beta1.controlplane.antrea.tanzu.vmware.com
4224-  name: v1beta1.system.antrea.io
4239-  name: v1beta1.system.antrea.tanzu.vmware.com
4254-  name: v1beta2.controlplane.antrea.io
4269-  name: v1beta2.controlplane.antrea.tanzu.vmware.com
```

Comment (edwardbadboy) [reply pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml:34]: @knabben Yes this is correct, thanks a lot.

File: pkg/v1/providers/infrastructure-vsphere/v0.7.10/ytt/overlay-windows.yaml
Comment (lzhecheng) [inline pkg/v1/providers/infrastructure-vsphere/v0.7.10/ytt/overlay-windows.yaml:335]: I think `-AntreaVersion "1.2.3"` should be `-AntreaVersion "v1.2.3"`

Comment (knabben) [reply pkg/v1/providers/infrastructure-vsphere/v0.7.10/ytt/overlay-windows.yaml:335]: @hxietkg is this breaking older versions?

Comment (knabben) [reply pkg/v1/providers/infrastructure-vsphere/v0.7.10/ytt/overlay-windows.yaml:335]: @hxietkg keeping "v1.2.3" here will break older Antrea versions?

**Comment by edwardbadboy** [comment]
Look good. Thanks!

File: pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml
Comment (knabben) [inline pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml:247]: enterpriseAntrea is disabled on CSR bootstrap phase

File: pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/upstream/antrea.yaml
Comment (knabben) [inline pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/upstream/antrea.yaml:4034]: enterpriseAntrea enabled on controller data config

**Comment by jayunit100** [comment]
overall this lgtm.  i think we should merge and iterate as needed.  theres no way to 100% verify this by eye.

File: pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml
Comment (edwardbadboy) [reply pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml:247]: May I know the reason to disable on CSR bootstrap phase?

**Comment by edwardbadboy** [comment]
Looks good. Thanks.

**Comment by vuil** [comment]
So are the review comments addressed? Please resolve them if so. 
Please review the last clustergen diffs:
https://github.com/vmware-tanzu/tanzu-framework/pull/979#issuecomment-963157841

File: pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml
Comment (knabben) [reply pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml:247]: This overlay commenting enterpriseAntrea is made on 0.13.3 and 0.11
https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/providers/ytt/vendir/cni/_ytt_lib/addons/packages/antrea/0.13.3/bundle/config/overlay/antrea_overlay.yaml#L198

**Comment by jayunit100** [comment]
we can hold this until other dependencies merge... but... I think we just want a soft approval at this point.  

The clustergendiff being empty *I think* is because this isn't integrated into the bom fully, so its effectively a no-op.

**Comment by knabben** [comment]
@vijaykatam can you PTAL again?

**Review by vijaykatam — APPROVED** [review]
LGTM. Please have a networking team member sign off as well.

**Comment by knabben** [comment]
> Looks good. Thanks.

@edwardbadboy is from Antrea team, could you approve the PR?

**Comment by edwardbadboy** [comment]
I didn't see an approve button or option. I'm not sure if it's because I don't have a privilege. From Antrea team's point of view I want to approve this patch.

**Pull request merged**
